src/lib/ruby-generator/control.js

```diff
@@ -4,12 +4,70 @@
  * @return {Blockly} Blockly defined Ruby generator.
  */
 export default function (Blockly) {
+    Blockly.Ruby.control_wait = function (block) {
+        const secs = Blockly.Ruby.valueToCode(block, 'DURATION', Blockly.Ruby.ORDER_NONE) || '0';
+        return `sleep(${secs})\n`;
+    };
+
     Blockly.Ruby.control_repeat = function (block) {
         const times = Blockly.Ruby.valueToCode(block, 'TIMES', Blockly.Ruby.ORDER_NONE) || '0';
         const branch = Blockly.Ruby.statementToCode(block, 'SUBSTACK') || '\n';
 
         return `${times}.times do\n${branch}end\n`;
     };
 
+    Blockly.Ruby.control_forever = function (block) {
+        const branch = Blockly.Ruby.statementToCode(block, 'SUBSTACK') || '\n';
+        return `forever do\n${branch}end\n`;
+    };
+
+    Blockly.Ruby.control_if = function (block) {
+        const operator = Blockly.Ruby.valueToCode(block, 'CONDITION', Blockly.Ruby.ORDER_NONE) || 'false';
+        const branch = Blockly.Ruby.statementToCode(block, 'SUBSTACK') || '\n';
+        return `if ${operator}\n${branch}end\n`;
+    };
+
+    Blockly.Ruby.control_if_else = function (block) {
+        const operator = Blockly.Ruby.valueToCode(block, 'CONDITION', Blockly.Ruby.ORDER_NONE) || 'false';
+        const branch = Blockly.Ruby.statementToCode(block, 'SUBSTACK') || '\n';
+        const branch2 = Blockly.Ruby.statementToCode(block, 'SUBSTACK2') || '\n';
+        return `if ${operator}\n${branch}else\n${branch2}end\n`;
+    };
+
+    Blockly.Ruby.control_wait_until = function (block) {
+        const operator = Blockly.Ruby.valueToCode(block, 'CONDITION', Blockly.Ruby.ORDER_NONE) || 'false';
+        return `wait until ${operator}\n`;
+    };
+
+    Blockly.Ruby.control_repeat_until = function (block) {
+        const operator = Blockly.Ruby.valueToCode(block, 'CONDITION', Blockly.Ruby.ORDER_NONE) || 'false';
+        const branch = Blockly.Ruby.statementToCode(block, 'SUBSTACK') || '\n';
+        return `until ${operator}\n${branch}end\n`;
+    };
+
+    // TODO 値を取得方法がわからない
+    Blockly.Ruby.control_stop = function () {
+        return '# wip: control_stop\n';
```

Comment:
scratch-blocks のコードを読んだ限り、おそらく STOP_OPTION で値を取得できます。

```suggestion
        const stop_option = block.getFieldValue('STOP_OPTION') || null;
        return `stop("${stop_option}")\n`;
```